app/facades/trip_facade.rb

```diff
@@ -6,78 +6,54 @@ def initialize(trip_params)
     @destination = trip_params[:destination]
     @trip = Trip.create
 
+
     # Add SidekiqWorker job here to build POIs and legs of trip
     # The worker will need the trip id/token and Google directions payload
     # Otherwise it will need to make a second Google API request
+
+    # @matt -- I think this is the right syntax -- Not sure
+
+    TripWorker.build_trip(_directions.steps)
   end
 
   def response
     return {
       data: {
         trip_token: trip.token,
-        places: [origin_data, destination_data],
-        legs: [legs_data]
+        places: [ origin_data, destination_data ],
+        legs: [ leg_data.json_with_id(1) ]
       }
     }
   end
 
   def origin_data
-    {
-       "location": {
-           "lat": _directions.start_lat,
-           "lng": _directions.start_lng
-       },
-       "name": _directions.start_address,
-       "state": Poi.poi_at_location(_directions.start_lat, _directions.start_lng)[0].state,
-       "population": Poi.population_at_location(_directions.start_lat, _directions.start_lng),
-       "weather": origin_weather
-   }
-  end
 
-  def origin_weather
-    weather_data = _weather_service.get(_directions.start_lat, _directions.start_lng)
+    origin_poi  = Poi.poi_at_location(_directions.start_lat, _directions.start_lng)
+                          .order(population: "DESC")
+                          .first
 
-    ForecastSerializer.new(weather_data).currently
+    PoiSerializer.new(origin_poi).to_json_with_timing(0)
   end
 
+
   def destination_data
-    {
-       "location": {
-           "lat": _directions.end_lat,
-           "lng": _directions.end_lng
-       },
-       "name": _directions.end_address,
-       "state": Poi.poi_at_location(_directions.end_lat, _directions.end_lng)[0].state,
-       "population": Poi.population_at_location(_directions.end_lat, _directions.end_lng),
-       "weather": destination_weather
-   }
-  end
 
-  def destination_weather
-    weather_data = _weather_service.get(_directions.end_lat, _directions.end_lng)
-    duration_hours = _directions.duration_hours
-    # Get weather of destination at X hours from now
-    ForecastSerializer.new(weather_data).hourly_in(duration_hours.round)
+    destination_poi  = Poi.poi_at_location(_directions.end_lat, _directions.end_lng)
+                          .order(population: "DESC")
+                          .first
```

Comment:
We should probably pull this into a new POI class method, maybe call it `Poi.most_populous_at(lat,lng)`

Reply:
Opened as #65